src/app/layout/app.menuitem.component.ts

```diff
@@ -65,7 +75,7 @@ export class AppMenuitemComponent implements OnInit, OnDestroy {
 
     key: string = "";
 
-    constructor(public layoutService: LayoutService, private cd: ChangeDetectorRef, public router: Router, private menuService: MenuService) {
+    constructor() {
         this.menuSourceSubscription = this.menuService.menuSource$.subscribe(value => {
```

Comment:
You don't need variable **menuSourceSubscription** to keep sub ref in order to unsub.
Angular core have new way of unsub -> using https://v16.angular.io/api/core/DestroyRef

Define variable
`private destroyRef = inject(DestroyRef);`
and on Observable use 
`.pipe(takeUntilDestroyed(this.destroyRef))`

Like so:
`this.menuService.menuSource$
      .pipe(takeUntilDestroyed(this.destroyRef))
      .subscribe(value => {
        Promise.resolve(null).then(() => {
          if (value.routeEvent) {
            this.active = (value.key === this.key || value.key.startsWith(this.key + '-'));
          } else {
            if (value.key !== this.key && !value.key.startsWith(this.key + '-')) {
              this.active = false;
            }
          }
        });
      });`

Reply:
`takeUntilDestroyed` is still dev preview but I do like the approach much better and have been using it in my personal projects.

Reply:
I do use this approach in my personal projects but because it's still dev preview I don't think it's good to put into a component library / template like this one.